### ISSUE TITLE
Possible false positive in `prefer-const`

### ISSUE THREAD

**Issue by daltones** [eslint/eslint#7117]
**The version of ESLint you are using:**
v3.5.0

**The problem you want to solve:**
In some cases you _need_ to use `let`, even if you don't reassign the variable. Them `prefer-const` raises a false positive.

**Your take on the correct solution to problem.**
Check if the variable is referenced inside a child scope before the assignment.

Example:
I need to use the value of `id` inside a function, but it will be assigned after.
```js
let id;
const handler = () => doSomethingWith(id);
id = createSomethingAndReturnId(handler);

// prefer-const will complain about id never being reassigned
```

**Comment by mysticatea** [comment]
Thank you for this issue.

You can rewrite the code:

```js
const handler = () => doSomethingWith(id);
const id = createSomethingAndReturnId(handler);
```

Or you can use [ignoreReadBeforeAssign](http://eslint.org/docs/rules/prefer-const#ignorereadbeforeassign) option for this.

**Comment by daltones** [comment]
You're right. Sorry, my fault.
